account_manager/src/validator/import.rs

```diff
@@ -0,0 +1,214 @@
+                .value_name("VALIDATOR_DIRECTORY")
+                .help(
+                    "The path where the validator directories will be created. \
+                    Defaults to ~/.lighthouse/validators",
+                )
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name(STDIN_PASSWORD_FLAG)
+                .long(STDIN_PASSWORD_FLAG)
+                .help("If present, read passwords from stdin instead of tty."),
+        )
+}
+
+pub fn cli_run(matches: &ArgMatches) -> Result<(), String> {
+    let keystore: Option<PathBuf> = clap_utils::parse_optional(matches, KEYSTORE_FLAG)?;
+    let keystores_dir: Option<PathBuf> = clap_utils::parse_optional(matches, DIR_FLAG)?;
+    let validator_dir = clap_utils::parse_path_with_default_in_home_dir(
+        matches,
+        VALIDATOR_DIR_FLAG,
+        PathBuf::new().join(".lighthouse").join("validators"),
+    )?;
+    let stdin_password = matches.is_present(STDIN_PASSWORD_FLAG);
+
+    ensure_dir_exists(&validator_dir)?;
+
+    let mut defs = ValidatorDefinitions::open_or_create(&validator_dir)
+        .map_err(|e| format!("Unable to open {}: {:?}", CONFIG_FILENAME, e))?;
+
+    // Collect the paths for the keystores that should be imported.
+    let keystore_paths = match (keystore, keystores_dir) {
+        (Some(keystore), None) => vec![keystore],
+        (None, Some(keystores_dir)) => {
+            let mut keystores = vec![];
+
+            recursively_find_voting_keystores(&keystores_dir, &mut keystores)
+                .map_err(|e| format!("Unable to search {:?}: {:?}", keystores_dir, e))?;
+
+            if keystores.is_empty() {
+                eprintln!("No keystores found in {:?}", keystores_dir);
+                return Ok(());
+            }
+
+            keystores
+        }
+        _ => {
+            return Err(format!(
+                "Must supply either --{} or --{}",
+                KEYSTORE_FLAG, DIR_FLAG
+            ))
+        }
+    };
+
+    eprintln!("WARNING: {}", KEYSTORE_REUSE_WARNING);
+
+    // For each keystore:
+    //
+    // - Obtain the keystore password, if the user desires.
+    // - Copy the keystore into the `validator_dir`.
+    // - Add the keystore to the validator definitions file.
+    //
+    // Skip keystores that already exist, but exit early if any operation fails.
+    let mut num_imported_keystores = 0;
+    for src_keystore in &keystore_paths {
+        let keystore = Keystore::from_json_file(src_keystore)
+            .map_err(|e| format!("Unable to read keystore JSON {:?}: {:?}", src_keystore, e))?;
+
+        eprintln!("");
+        eprintln!("Keystore found at {:?}:", src_keystore);
+        eprintln!("");
+        eprintln!(" - Public key: 0x{}", keystore.pubkey());
+        eprintln!(" - UUID: {}", keystore.uuid());
+        eprintln!("");
+        eprintln!(
+            "If you enter the password it will be stored in {} so that it is not required \
```

Comment:
I think we should emphasise here that it is stored as plain text, for users who aren't really sure what they're doing and just following instructions.

Reply:
Great catch!